src/main.rs

```diff
@@ -149,7 +162,7 @@ async fn main() -> Result<()> {
                     }
                 }
 
-                push::push(&policy, uri, docker_config, sources).await?;
+                push::push(&policy, uri.as_str(), docker_config, sources).await?;
```

Comment:
```suggestion
                push::push(&policy, &uri, docker_config, sources).await?;
```

It's personal taste again, I favour `&uri`, but whatever you prefer :)